### PR TITLE
Month Selection Component

### PR DESCRIPTION
**Pull request by pedrops164**
Changed the Month Selection Component to receive functions onMonthChange and onYearChange as properties. Also added next and previous month buttons, and a text indicanting the current month.
I didn't understand how I could represent the month/year as properties, since I don't know how I would change the value of those properties once the user clicked the next or previous buttons.

### CODE REVIEW

File: components/MonthSelection.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react'
```
Comment (ruioliveira02) [inline components/MonthSelection.js:1]:
```suggestion

```
Remove this as it won't be needed

File: components/MonthSelection.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const MonthSelection = ({onMonthChange, onYearChange}) => {
+    const [month, setMonth] = useState(0) //estado inicial é janeiro
+    const [year, setYear] = useState(2020)
```
Comment (ruioliveira02) [inline components/MonthSelection.js:5]:
```suggestion
```

Component should be stateless

File: components/MonthSelection.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            case 3: return "Abril";
+            case 4: return "Maio";
+            case 5: return "Junho";
+            case 6: return "Julho";
+            case 7: return "Agosto";
+            case 8: return "Setembro";
+            case 9: return "Outubro";
+            case 10: return "Novembro";
+            case 11: return "Dezembro";
+        }
```
Comment (ruioliveira02) [inline components/MonthSelection.js:48]:
```suggestion
        const names = ["Janeiro", "Fevereiro", "Março", "Abril", "Maio", "Junho",
         "Julho",   "Agosto", "Setembro", "Outubro", "Novembro", "Dezembro"];
         
         return names[month];
        
```

This is a lot cleaner

File: components/MonthSelection.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react'
+
+const MonthSelection = ({onMonthChange, onYearChange}) => {
```
Comment (ruioliveira02) [inline components/MonthSelection.js:3]:
```suggestion
const MonthSelection = ({onMonthChange, onYearChange, month, year}) => {
```

The year and month should be received as properties

File: components/MonthSelection.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    //skips to the next month, can also handle if it is December
+    const nextMonth = () => {
+        if (month===11) {
+            //if it is December (11), we set the month back to January (0), and we change year.
+            setMonth(0)
+            setYear(year + 1)
+            onYearChange //calls function onYearChange??
+        }
+        else setMonth(month + 1)
```
Comment (ruioliveira02) [inline components/MonthSelection.js:15]:
```suggestion
        else 
           onMonthChange (month + 1);
```

File: components/MonthSelection.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    const [month, setMonth] = useState(0) //estado inicial é janeiro
+    const [year, setYear] = useState(2020)
+
+    //skips to the next month, can also handle if it is December
+    const nextMonth = () => {
+        if (month===11) {
+            //if it is December (11), we set the month back to January (0), and we change year.
+            setMonth(0)
+            setYear(year + 1)
+            onYearChange //calls function onYearChange??
```
Comment (ruioliveira02) [inline components/MonthSelection.js:13]:
```suggestion
            onMonthChange(0);
            onYearChange(year + 1);
```

Functions passed as arguments still have to be called using parenthesis. This also applies for the previousMonth function.

File: components/MonthSelection.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        else setMonth(month + 1)
+
+        onMonthChange //calls function onMonthChange??
```
Comment (ruioliveira02) [inline components/MonthSelection.js:17]:
```suggestion
```

File: components/MonthSelection.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            case 9: return "Outubro";
+            case 10: return "Novembro";
+            case 11: return "Dezembro";
+        }
+    }
+    
+  return (
+    <div>
+        <button onClick={previousMonth}>&larr;</button>
+        <text>{showMonth()}</text>
```
Comment (ruioliveira02) [inline components/MonthSelection.js:54]:
```suggestion
        <h2>{showMonth()}</h2>
```

Text tags are not a thing.

**Review by ruioliveira02 — CHANGES_REQUESTED** [review]

**Review by pedrops164 — COMMENTED** [review]
In order to change the months with the scroll wheel, while maintaining the component stateless, should the component receive the scroll wheel Y and the function to change this value?

**Comment by ruioliveira02** [comment]
> In order to change the months with the scroll wheel, while maintaining the component stateless, should the component receive the scroll wheel Y and the function to change this value?

Let's not do that now.

**Review by ruioliveira02 — CHANGES_REQUESTED** [review]
Please move the file ```components/MonthSelection.js``` to ````components/MonthSelection/index.jsx````

In regards to styling, look for similar to [SEI](https://seium.org). But for now let's wait for other components to be developed to create a uniform design

**Comment by ruioliveira02** [comment]
Same reason as #7. Thank you for your contribution regardless :pray:

**Pull request closed without merging**